homeassistant/components/filter/sensor.py

```diff
@@ -177,22 +179,24 @@ async def async_setup_platform(
     await async_setup_reload_service(hass, DOMAIN, PLATFORMS)
 
     name = config.get(CONF_NAME)
+    unique_id = config.get(CONF_UNIQUE_ID)
     entity_id = config.get(CONF_ENTITY_ID)
 
     filters = [
         FILTERS[_filter.pop(CONF_FILTER_NAME)](entity=entity_id, **_filter)
         for _filter in config[CONF_FILTERS]
     ]
 
-    async_add_entities([SensorFilter(name, entity_id, filters)])
+    async_add_entities([SensorFilter(name, unique_id, entity_id, filters)])
 
 
 class SensorFilter(SensorEntity):
     """Representation of a Filter Sensor."""
 
-    def __init__(self, name, entity_id, filters):
+    def __init__(self, name, unique_id, entity_id, filters):
         """Initialize the sensor."""
         self._name = name
+        self._unique_id = unique_id
```

Comment:
A minor thing, why not?
```suggestion
        self._attr_unique_id = unique_id
```

Reply:
Good point :)